reco_utils/recommender/vae/sparse_vae_2.py

```diff
@@ -0,0 +1,194 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
```

Comment:
is this file needed? there are similar functions in reco_utils

Reply:
indeed in reco_utils exist similar functions but for the scope of our project and because of the different preprocessing we needed to do some changes in those functions. That is why we uploaded a new file as sparse_vae_2.py.

Reply:
Evgenia, several questions:

1) what are the differences between this AffinityMatrix and the original one? 
2) is there a way to extend the original one to not repeat all the code?
3) Something that we try to do is https://github.com/Microsoft/Recommenders/wiki/Coding-Guidelines#do-not-repeat-yourself, would it be possible to at least share part of the code?

Reply:
Hi Miguel,
1. In original approach it takes the whole dataset, and converts it to the sparse matrix of size MxN (where M number of unique users, and N number of unique items).
However, when we split our dataset into separate train/validation/test users, there is no guarantee that the number of unique items will remain the same, hence we need to control this number of items within all sparse matrices. In order to control it, we changed original code a little bit. 
To track the changes, just look for comments: "# LSECHANGE:". There are only 4 lines of code changed so far.

2. The main thing that we have changed in the code, is that instead of obtaining a list of unique items from a provided dataset, we are giving this list as an input (because sometimes list obtained from a dataframe is not what we need). We think it is possible to extend the original code, but we will need to keep additional input (list of unique items) anyways. The code might look as following: if such list is provided, the code should use it for sparse matrix, if not provided, it should obtain a list from a dataframe.

3. Sorry, the question is not really clear for us. Can you clarify it?

Reply:
Also, we just found out that binarize_matrix function has analogical function in the reco_utils, hence it is going to be removed and replaced by binarize function of reco_utils - python_utils.py

Reply:
> The code might look as following: if such list is provided, the code should use it for sparse matrix, if not provided, it should obtain a list from a dataframe.

I think that would be ok, can you please try?

question 3 is basically doing what you proposed :-)